File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        #Key = number
        #Value = index
        ans = {}

        for i in range(len(nums)):
            diff = target - nums[i]
    
            if diff in ans:
                return [ans.get(diff), i]

            ans[nums[i]] = i

            
            
            
            
            


        

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int fp = 0;
        int lp = nums.length - 1;

        while (lp >= fp) {
            int mid = (int) lp + (int) Math.floor((fp - lp)/2);
            if (nums[mid] > target) {
                lp = mid - 1;
            } else if (nums[mid] < target) {
                fp = mid + 1;
            } else {
                return mid;
            }


        }

        return -1;
    }
}
